log.go

```diff
@@ -0,0 +1,212 @@
+package stripe
```

Comment:
I ended up moving all log-related constructs to a new file in the top-level package so that it's easier to find log-related things.